Cw2 2

alkor63 opened the pull request.

alkor63 committed:
2nd commit
create new classes

alkor63 committed:
2nd commit
create new classes
перенес коды ежедневника

alkor63 committed:
2nd commit
create new classes
перенес коды ежедневника
добавил код аналиа вводимого текста
(см. TextScanner в пакете TextCounter
там свой main)

alekseykoshelev commented on src/CourseWork2code/DailyTask.java:
```diff
@@ -0,0 +1,28 @@
+
+public class DailyTask extends Task {
+    //        implements Comparable
+    private final LocalDateTime dateTime;
```
Такие же поля есть в родительском классе, тут их можно не дублировать.

alkor63 replied:
Почему-то не получается, если не добавлять дату в качестве аргумента.  
А по логике задачи я дату не передаю снаружи, а присваиваю текущую дату при создании объекта

alekseykoshelev commented on src/CourseWork2code/IncorrectArgumentException.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String toString() {
+        return "*** IncorrectArgumentException - некорректный титул задачи ***";
```
В таком случае, раз тут поле есть, можно его сюда добавить)

alkor63 replied:
Здесь речь про титул задачи = null или пустышка. Решил, что вывод пустоты информации не добавит

alekseykoshelev commented on src/CourseWork2code/MonthlyTask.java:
```diff
@@ -0,0 +1,29 @@
+
+    @Override
+    public boolean appearsIn(LocalDate inputDate, LocalDate taskDate) {
+        if (inputDate.getDayOfMonth() == taskDate.getDayOfMonth())
```
Условие правильно, можно упростить `return inputDate.getDayOfMonth() == taskDate.getDayOfMonth();`

alkor63 replied:
+

alekseykoshelev commented on src/CourseWork2code/OneTimeTask.java:
```diff
@@ -0,0 +1,27 @@
+import java.time.LocalDateTime;
+
+public class OneTimeTask extends Task {
+    private final LocalDateTime dateTime;
```
Аналогичный момент по полям.

alekseykoshelev commented on src/Main.java:
```diff
@@ -1,5 +1,79 @@
+            System.out.println("id = " + tasK.getKey() + " task = " + tasK.getValue());
+    }
+
+    private static void printMenu1() {
```
А почему в конце 1?)

alkor63 replied:
Исторически) Когда начинал писать этот код уже понимал, что у меня будет 3 меню, но про разные классы еще не думал

alekseykoshelev commented on src/CourseWork2code/TaskService.java:
```diff
@@ -0,0 +1,166 @@
+
+    public static void inputTask(Scanner scanner, Map<Integer, Task> tasks) {
+        Scanner scanner1 = new Scanner(System.in);
+        Scanner scanner2 = new Scanner(System.in);
```
Получается в этом методе 3 сканнера, один это аргумент, 2 ты создал. А зачем так много?

alkor63 replied:
Перестраховка. Не знаю, что у сканера под капотом, но убедился, что текст, числа и, например, даты лучше вводить через разные сканеры.
scanner1 убрал.

alekseykoshelev commented on src/CourseWork2code/TaskService.java:
```diff
@@ -0,0 +1,166 @@
+                break;
+            default: {
+                System.out.println("ТИП задачи указан невнрно, ты всё сломал !");
+                taskTyp = Type.valueOf("ЛИЧНАЯ");
```
Вариант интересный, но ведь он не будет работать. Можно же использовать `Type.PERSONAL`

alkor63 replied:
+

alekseykoshelev commented on src/CourseWork2code/TaskService.java:
```diff
@@ -0,0 +1,166 @@
+public class TaskService {
+    static Set<Task> taskSet = new HashSet<>();
+
+    public static void inputTask(Scanner scanner, Map<Integer, Task> tasks) {
```
Не совсем понимаю почему в каждом методе есть аргумент типа `Map<Integer, Task> tasks`. Ведь тут у тебя есть поля такого же типа, ты можешь к нему обращаться не передавая как аргумент. Поле, кстати, можно сделать private

alkor63 replied:
Вот здесь я недопонял. Разве у меня есть внутри этого метода (и тех, сто в этом же классе) к мапе, созданной в main без передачи мапы как аргумент? 
какое поле private? модификатор метода в 9 с троке? а метод при этом будет доступен из main ?

alekseykoshelev commented on src/CourseWork2code/TaskService.java:
```diff
@@ -0,0 +1,166 @@
+        System.out.println("Вот что вы навводили:\n" + task9);
+    }
+
+    public static void updateDescription(Map<Integer, Task> tasks) {
```
Сюда, как вариант, можно передавать сканнер аргументом.

alekseykoshelev commented on src/CourseWork2code/TaskService.java:
```diff
@@ -0,0 +1,166 @@
+        try {
+            checkId(tasks, id);
+            Task task = tasks.get(id);
+            taskSet.add(task);
```
По поводу аргумента я понял, но в таком случае лучше по другому назвать переменную, иначе вообще не понятно, что она хранит удаленные задачи.

alkor63 replied:
renamed to archiveRemovedTaskSet

alekseykoshelev commented on src/TextCounter/TextScanner.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    public static HashMap<String, Integer> makeWordList(String inputText) {
+        Scanner scan = new Scanner(inputText);
```
В целом верно, но можно чуть проще. Через метод `inputText.split(" ")`

alekseykoshelev commented on src/TextCounter/TextScanner.java:
```diff
@@ -0,0 +1,84 @@
+
+    public static HashMap<String, Integer> makeWordList(String inputText) {
+        Scanner scan = new Scanner(inputText);
+        HashMap<String, Integer> listOfWords = new HashMap<String, Integer>();
```
Ну это не совсем list, скорее map)

alkor63 replied:
map с понятным названием Список )

alekseykoshelev commented on src/TextCounter/TextScanner.java:
```diff
@@ -0,0 +1,84 @@
+            } else {
+                countWord = listOfWords.get(word) + 1; //get current count and increment
+                //now put the new value back in the HashMap
+                listOfWords.remove(word); //first remove it (can't have duplicate keys)
```
Можно не удалять, можно сразу делать put. значение просто перезапишется

alkor63 replied:
+

alkor63 committed:
3 commit
внес некоторые исправления

alkor63 committed:
4 commit
убрал аргументы из большинства методов TaskService

alkor63 committed:
5 commit
в TaskService оставил один сканер

alkor63 committed:
6 commit
убрал всё лишнее из IncorrectArgumentException

alkor63 committed:
7 commit
"причесал" всех потомков Task (убрал id & dataTime)

alkor63 committed:
7 commit
"причесал" всех потомков Task (убрал id & dataTime)

alkor63 committed:
7 commit
"причесал" всех потомков Task (убрал id & dataTime)
LiveCoding к билету 5 см. в пакете Number5
